src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using StreamJsonRpc;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
```

Comment:
Why do we have this type in the core Microsoft.VisualStudio.LanguageServices.dll instead of keeping it in the LiveShare DLL? (This I think is what's forcing the extra project reference to be added, no?)

I can imagine your answer is "well, one day this is just how things work, so mind as well start it now."

Reply:
3 reasons
1.  Side by side the existing c# lsp client, so we can easily migrate features to it.  It's not necessarily tied to liveshare.
2.  My goal is to delete the whole liveshare project all together.
3.  This project already has the VS LSP client dependencies.